src/benchmarks/micro/corefx/System.Globalization/StringSearch.cs

```diff
@@ -90,5 +105,139 @@ public void Setup()
 
         [Benchmark]
         public int LastIndexOf_Word_NotFound() => Options.CultureInfo.CompareInfo.LastIndexOf(_value, "word", Options.CompareOptions);
+
+        private static bool ContainsHighChar(string text) => text.Any(c => c > 0x80 || HighChars[c]);
+
+        // source: https://github.com/dotnet/coreclr/blob/68ff240063fc2ddb9b03275ae5d5063a09d38ace/src/utilcode/util_nodependencies.cpp#L856-L993
```

Comment:
This array gets tweaked from time to time, and it is unfortunate to have it hardcoded in yet another place.

It may be better to implement the sanity check against "known safe" range that is unlikely to change, e.g. `a-zA-Z0-9 .`.

Reply:
good point, I've implemented that